app/forms/fields.py

```diff
@@ -163,24 +164,61 @@ def get_integer_field(answer, label, guidance, error_messages):
 
     mandatory_or_optional = get_mandatory_validator(answer, answer_errors)
 
-    if answer['type'] == 'Integer':
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(maximum=9999999999, messages=answer_errors),
-        ]
-    elif answer['type'] == 'Percentage':
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(minimum=0, maximum=100, messages=answer_errors),
-        ]
-    else:
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(minimum=0, maximum=9999999999, messages=answer_errors),
-        ]
+    max_decimals = answer.get('decimal_places', 0)
+    if max_decimals > MAX_DECIMAL_PLACES:
+        max_decimals = MAX_DECIMAL_PLACES
 
-    return CustomIntegerField(
-        label=label,
-        description=guidance,
-        validators=validate_with,
-    )
+    min_value = get_schema_defined_max_min_value(answer.get('min_value'), answer_store) or 0
+
+    max_value = get_schema_defined_max_min_value(answer.get('max_value'), answer_store) \
+        or get_max_min_system_limits('max', max_decimals)
+
+    if min_value > max_value:
+        raise Exception('min_value: ' + str(min_value) + ' > max_value: ' + str(max_value) +
+                        ' for answer id: ' + answer['id'])
+
+    validate_with = mandatory_or_optional + [
+        NumberCheck(answer_errors['INVALID_NUMBER']),
+        NumberRange(minimum=min_value, maximum=max_value, messages=answer_errors),
+        DecimalPlaces(max_decimals=max_decimals, messages=answer_errors),
+    ]
+
+    if max_decimals > 0:
+        return CustomDecimalField(
+            label=label,
+            validators=validate_with,
+            description=guidance,
+        )
+    else:
+        return CustomIntegerField(
+            label=label,
+            validators=validate_with,
+            description=guidance,
+        )
+
+
+def get_schema_defined_max_min_value(definition, answer_store):
+    if definition:
+        if 'value' in definition:
+            return definition['value']
+        else:
+            answer_id = definition['answer_id']
+            for answer in answer_store.answers:
```

Comment:
`answer_store.filter(answer_id=answer_id)` will return a list of answers that match an answer_id and should be used.